src/Compilers/CSharp/Portable/Syntax/SyntaxFactory.cs

```diff
@@ -1543,14 +1543,18 @@ public static IdentifierNameSyntax IdentifierName(string name)
         /// <summary>
         /// Create a new syntax tree from a syntax node.
         /// </summary>
-        public static SyntaxTree SyntaxTree(SyntaxNode root, ParseOptions? options = null, string path = "", Encoding? encoding = null)
-        {
-            return CSharpSyntaxTree.Create((CSharpSyntaxNode)root, (CSharpParseOptions?)options, path, encoding);
-        }
+        public static SyntaxTree SyntaxTree(SyntaxNode root, ParseOptions? options, string path, Encoding? encoding)
+            => SyntaxTree(root, options, path, encoding, SourceHashAlgorithm.Sha1);
+
+        /// <summary>
+        /// Create a new syntax tree from a syntax node.
+        /// </summary>
+        public static SyntaxTree SyntaxTree(SyntaxNode root, ParseOptions? options = null, string path = "", Encoding? encoding = null, SourceHashAlgorithm checksumAlgorithm = SourceHashAlgorithm.Sha1)
+            => CSharpSyntaxTree.Create((CSharpSyntaxNode)root, (CSharpParseOptions?)options, path, encoding, checksumAlgorithm);
 
 #pragma warning disable RS0026 // Do not add multiple public overloads with optional parameters
 #pragma warning disable RS0027 // Public API with optional parameter(s) should have the most parameters amongst its public overloads.
-
+#pragma warning disable RS0030 // Do not used banned APIs
```

Comment:
Why is this API allowed to call a banned API?

Reply:
And which API is it?

Reply:
This API is also banned. Calling `CSharpSyntaxTree.ParseText`.